File: src/main/java/me/mrscopes/beans/events/JoinListener.java
package me.mrscopes.beans.events;

import me.mrscopes.beans.Beans;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import net.kyori.adventure.text.minimessage.tag.resolver.Placeholder;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

public class JoinListener implements Listener {
    @EventHandler
    public void onJoin(PlayerJoinEvent event) {
        event.joinMessage(MiniMessage.miniMessage().deserialize("<player> joined the server.", Placeholder.component("player", Component.text(event.getPlayer().getName()))));
        Beans.getDiscord().getServerChat().sendMessage(String.format("✅ **%s** joined the server.", event.getPlayer().getName())).queue();
    }
}

File: src/main/kotlin/me/mrscopes/beans/events/JoinListener.kt
package me.mrscopes.beans.events

import me.mrscopes.beans.Beans
import me.mrscopes.beans.color
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class JoinListener : Listener {
    @EventHandler
    fun onJoin(event: PlayerJoinEvent) {
        val player = event.player
        event.joinMessage("&8[&a+&8] &7${player.name}".color())
        Beans.discord.serverChat.sendMessage("✅ **${player.name}** joined the server.").queue()
    }
}
